stable_baselines3/her/her.py

```diff
@@ -0,0 +1,495 @@
+                gradient_steps = self.gradient_steps if self.gradient_steps > 0 else rollout.episode_timesteps
+                self.train(batch_size=self.batch_size, gradient_steps=gradient_steps)
+
+        callback.on_training_end()
+
+        return self
+
+    def collect_rollouts(
+        self,
+        env: VecEnv,
+        callback: BaseCallback,
+        n_episodes: int = 1,
+        n_steps: int = -1,
+        action_noise: Optional[ActionNoise] = None,
+        learning_starts: int = 0,
+        replay_buffer: Union[ReplayBuffer, HerReplayBuffer] = None,
+        log_interval: Optional[int] = None,
+    ) -> RolloutReturn:
+        """
+        Collect experiences and store them into a ReplayBuffer.
+
+        :param env: The training environment
+        :param callback: Callback that will be called at each step
+            (and at the beginning and end of the rollout)
+        :param n_episodes: Number of episodes to use to collect rollout data
+            You can also specify a ``n_steps`` instead
+        :param n_steps: Number of steps to use to collect rollout data
+            You can also specify a ``n_episodes`` instead.
+        :param action_noise: Action noise that will be used for exploration
+            Required for deterministic policy (e.g. TD3). This can also be used
+            in addition to the stochastic policy for SAC.
+        :param learning_starts: Number of steps before learning for the warm-up phase.
+        :param replay_buffer:
+        :param log_interval: Log data every ``log_interval`` episodes
+        :return:
+        """
+
+        episode_rewards, total_timesteps = [], []
+        total_steps, total_episodes = 0, 0
+
+        assert isinstance(env, VecEnv), "You must pass a VecEnv"
+        assert env.num_envs == 1, "OffPolicyAlgorithm only support single environment"
+
+        if self.model.use_sde:
+            self.actor.reset_noise()
+
+        callback.on_rollout_start()
+        continue_training = True
+
+        while total_steps < n_steps or total_episodes < n_episodes:
+            done = False
+            episode_reward, episode_timesteps = 0.0, 0
+
+            while not done:
+                # concatenate observation and (desired) goal
+                observation = self._last_obs
+                self._last_obs = ObsDictWrapper.convert_dict(observation)
+
+                if self.model.use_sde and self.model.sde_sample_freq > 0 and total_steps % self.model.sde_sample_freq == 0:
+                    # Sample a new noise matrix
+                    self.actor.reset_noise()
+
+                # Select action randomly or according to policy
+                self.model._last_obs = self._last_obs
+                action, buffer_action = self._sample_action(learning_starts, action_noise)
+
+                # Perform action
+                new_obs, reward, done, infos = env.step(action)
+
+                done = done if episode_timesteps < self.max_episode_length else False
```

Comment:
you should comment that line for now

Reply:
or handle it as in https://github.com/DLR-RM/stable-baselines3/compare/feat/remove-timelimit